src/TwigFractal.php

```diff
@@ -2,50 +2,194 @@
 
 /**
  * @file
- * Contains \Drupal\twig_fractal\TwigFractal.
+ * Contains \Drupal\twig_fractal\Node\Render.
  */
 
-namespace Drupal\twig_fractal;
+namespace Drupal\twig_fractal\Node;
 
-use Drupal\twig_fractal\TokenParser\Render;
-use Drupal\twig_fractal\NodeVisitors\Attributes;
-use Twig\Extension\AbstractExtension;
+use Drupal\Core\Template\Attribute;
+use Twig\Node\Expression\AbstractExpression;
+use Twig\Node\IncludeNode;
+use Twig\Compiler;
+use Twig_Node_Expression;
+use Twig_Node_Include;
 
 /**
- * Registers a new `render` token parser to make Fractal components reusable in Drupal.
+ * Compiles `render` nodes.
  *
- * @see \Drupal\twig_fractal\TokenParser\Render
+ * Unlike `include` nodes, the default values for template variables are
+ * automatically retrieved from the corresponding component definition
+ * file in the Fractal component library and added to the compiled Twig PHP
+ * template string.
  *
- * @implements Twig\Extension\ExtensionInterface
+ * The resulting template variables consist of:
+ *
+ * 1. the default `context` properties in the component's definition file,
+ *    which may be overridden by
+ * 2. the `context` properties of the requested variants (if any) in the
+ *    component's definition file, which may be overridden by
+ * 3. the variables passed to the `render` tag itself.
  */
-class TwigFractal extends AbstractExtension {
-
-  /**
-   * Adds a new Render token parser instance to the list of parsers.
-   *
-   * @return array
-   *   The token parsers.
-   */
-  public function getTokenParsers(): array {
-    return [
-      new Render(),
-    ];
-  }
-
-  public function getNodeVisitors() {
-    return [
-      new Attributes(),
-    ];
-  }
-
-  /**
-   * Returns the extension name.
-   *
-   * @return string
-   *   The extension name.
-   */
-  public function getName(): string {
-    return 'twig_fractal';
-  }
+class Render extends IncludeNode {
```

Comment:
Please note you duplicated the methods (not sure if you also made adjustments) from the actual render class https://github.com/ipwa/twig_fractal/blob/master/src/Node/Render.php#L60

could you please double check, whether your adjustments in this file are correct? Also the class and filename do not align anymore.